activesupport/lib/active_support/core_ext/date_time/calculations.rb

```diff
@@ -137,12 +137,18 @@ def beginning_of_minute
   end
   alias :at_beginning_of_minute :beginning_of_minute
 
-  # Returns a new DateTime representing the end of the minute (hh:mm:59).
+  # Returns a new DateTime representing the end of the minute (hh:mm:59.999999).
   def end_of_minute
-    change(:sec => 59)
+    change(:sec => 59, :usec => 999999)
   end
   alias :at_end_of_minute :end_of_minute
 
+  # Returns a new DateTime representing the end of the second (hh:mm:ss.999999).
+  def end_of_second
```

Comment:
Yes. I was trying to follow the convention. If that's a bad choice I can remove it. What do you think?